docs/architecture/adr-054-crypto-encoding-2.md

```diff
@@ -0,0 +1,67 @@
+# ADR 054: Crypto encoding (part 2)
+
+## Changelog
+
+\*2020-2-27: Created
+
+## Context
+
+Amino has been a pain point of many users in the ecosystem. While Tendermint does not suffer greatly from the performance degradation introduced by amino, we are making an effort in moving the encoding format to a widely adopt format, [Protocol Buffers](https://developers.google.com/protocol-buffers). With this migration a new standard is needed for the encoding of keys. This will cause ecosystem wide breaking changes.
+
+Currently amino encodes keys as `<PrefixBytes> <Length> <ByteArray>`.
+
+> With keys becoming a [oneof type](https://developers.google.com/protocol-buffers/docs/proto3#oneof) we have the option of removing the prefix bytes.
+
+## Options
+
+If there are more options please leave a comment and we can discuss it.
+
+1. Remove backwards compatibility and go with `oneof` proto encoding of keys.
+
+- This will be a major breaking change in the ecosystem.
+- Will need a migration script to be released in conjunction with the release which will have this change.
+
+> I am leaning towards the first option and go through the with headache in order to entirely remove the amino overhead
+
+2. Keep the amino encoding format `<PrefixBytes> <Length> <ByteArray>` and protobuf encode the key after appending the prefix bytes.
+
+- This may cause issues with bech32 as it will exceed the maximum length permitted by the libraries
```

Comment:
I'm pretty sure we've already experienced problems with this approach (at least for bechified pub keys).